packages/sanity/src/core/form/members/object/fields/ObjectField.tsx

```diff
@@ -114,6 +90,87 @@ export const ObjectField = function ObjectField(props: {
     [onChange, member, pendingValue]
   )
 
+  return (
+    <FormCallbacksProvider
+      onFieldGroupSelect={onFieldGroupSelect}
+      onChange={handleChange}
+      onSetFieldSetCollapsed={onSetFieldSetCollapsed}
+      onPathOpen={onPathOpen}
+      onSetPathCollapsed={onSetPathCollapsed}
+      onPathBlur={onPathBlur}
+      onPathFocus={onPathFocus}
+    >
+      <FieldProvider
+        actions={field.actions}
+        documentId={documentId}
```

Comment:
Might be missing something obvious, but I don't quite understand why we need to wrap each node with this provider. Do we support the ability to override field actions for subtrees? If not - can't this info be read from the root context in the FormField component?

Reply:
The provider resolves the field actions for each individual field. There fields behave differently (can be also be hidden) depending on the field's path.

Reply:
Could this logic not be moved to the compoent that renders the field actions? I.e. is there a reason we have to resolve at this stage, and not in the field component itself (as long as it knows its path, that should be possible, no?)

Reply:
The reason is so implementers of field components don't have to figure out how to resolve the actions, but get them automatically (similar to presence).